src/ctx.cpp

```diff
@@ -71,36 +76,20 @@ void pj_set_ctx( projPJ pj, projCtx ctx )
     }
     for( const auto &alt: pj->alternativeCoordinateOperations )
     {
-        pj_set_ctx(alt.pj, ctx);
+        proj_assign_context(alt.pj, ctx);
     }
-}
-
-/************************************************************************/
-/*                        proj_assign_context()                         */
-/************************************************************************/
 
-/** \brief Re-assign a context to a PJ* object.
- *
- * This may be useful if the PJ* has been created with a context that is
- * thread-specific, and is later used in another thread. In that case,
- * the user may want to assign another thread-specific context to the
- * object.
- */
-void proj_assign_context( PJ* pj, PJ_CONTEXT* ctx )
-{
-    pj_set_ctx( pj, ctx );
 }
 
 /************************************************************************/
 /*                          createDefault()                             */
 /************************************************************************/
 
-projCtx_t projCtx_t::createDefault()
+pj_ctx pj_ctx::createDefault()
```

Comment:
shouldn't that be PJ_CONTEXT ? can be for a follow-up task

Reply:
It could be, but I decided to follow Thomas' original plan described at the top of `proj.h`:

https://github.com/OSGeo/PROJ/blob/c560b7957664c32e2465e8425abaccc5a6b2607d/src/proj.h#L41-L53

The `PJ` typedef has the `PJconsts` struct behind so having the `PJ_CONTEXT` typedef refer to the `pj_ctx` struct follows the same recipe.

I have no preference for one or the other, but that was my reasoning.